RTL/translation fixes

OmarIthawi opened the pull request:
Changes:

 - Explicit SCSS direciton in RTL to fix multiple direction issues
 - Make the submission confirmation message translatable

TODO:
 - [x] Have it tested by @devalih
 - [x] Rebase to master so it can be merged


Before:
-----
![ora-rtl-before](https://cloud.githubusercontent.com/assets/645156/6017235/9a120d52-ab94-11e4-9156-5c9a6cbe6c75.png)


After:
-----
![ora-rtl-after](https://cloud.githubusercontent.com/assets/645156/6017237/9ca93e00-ab94-11e4-9a09-d8280c619510.png)

edx-webhook commented:
Thanks for the pull request, @OmarIthawi! I've created [OSPR-365](https://openedx.atlassian.net/browse/OSPR-365) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams. 

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ('this must be merged by XX date', and why that is)
- partner information ('this is a course on edx.org')
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the Github pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

We can't start reviewing your pull request until you've added yourself to the [AUTHORS](https://github.com/edx/edx-ora2/blob/master/AUTHORS) file. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

sarina commented:
Hey @OmarIthawi - so happy to see these RTL fixes coming in! Please let me know when this is ready to be reviewed by us, or if you need some help.

OmarIthawi commented:
Hi @sarina,

I'm so happy too to be a contributor to this repo :),
@devalih have reviewed it and it should be OK. I have also rebased this branch to the latest master to fix conflicts in the minfied/compiled assets, so now it should be mergable.

sarina commented on openassessment/xblock/static/js/src/lms/oa_response.js:
```diff
@@ -433,10 +433,7 @@ OpenAssessment.ResponseView.prototype = {
-            "You're about to submit your response for this assignment. " +
-            "After you submit this response, you can't change it or submit a new response."
-        );
+        var msg = gettext("You're about to submit your response for this assignment. After you submit this response, you can't change it or submit a new response.");
```
You should be able to keep this in the broken-line form, within gettext. This keeps it readable on Github! Try:
```javascript
var msg = gettext(
    "You're about to submit your response for this assignment. " +
    "After you submit this response, you can't change it or submit a new response."
)
```

OmarIthawi replied:
@sarina Well the issue is that django translation extractor don't understand the former form i.e. it uses hacks and treats JavaScript as a C language. That's why you can't find this on transifex. Check this for detailed explanation: http://stackoverflow.com/a/24579117/161278

sarina replied:
Ah, this makes sense. I think it would be good to add a comment above this line so future developers don't make the same mistake.

OmarIthawi replied:
I've added the comment so other devs know why.

sarina commented:
@talbs help reviewing? :)

sarina commented:
@OmarIthawi since this is a bugfix change, can you specifically call out what you fixed? Screenshots are best. If that's not possible, please point out where the problems were with the RTL stylings so we can verify the fix.

talbs commented on openassessment/xblock/static/sass/oa/utilities/_shame.scss:
```diff
@@ -4,6 +4,7 @@
 // NOTES: use for any bad-form/orphaned scss that knowingly violate edX FED architecture/standards (see - http://csswizardry.com/2013/04/shame-css)
 
 .openassessment {
+  @include direction();
```
Looking at the documentation for this mixin (https://github.com/edx/edx-ora2/blob/fd6da7dc8d314f9a79b944c1a3db7b1b6e88e471/openassessment/xblock/static/sass/xb/libs/neat/grid/_direction-context.scss#L7), it looks like an argument should be passed in with it (e.g. right-to-left or left-to-right). 

Is there a reason why this doesn't have an argument? How are you passing in the RTL or LTR setting?

talbs replied:
Also, is there a specific reason why this is in the _shame.scss Sass partial? Do you need at the end of the compilation? Or something else?

OmarIthawi replied:
I moved it from `_shame.scss`, now I know why there's shame :+1: 

And for the mixin, the one I'm using is this: https://github.com/edx/edx-ora2/blob/master/openassessment/xblock/static/sass/vendor/bi-app/_mixins.scss#L266

and it's just a simple `@include direction();` and nothing more.

OmarIthawi replied:
Just to make it clear, I'm not sure why the CSS direction isn't being inherited from the lms styles. But this fixed it (see the screenshots above in https://github.com/edx/edx-ora2/pull/664#issue-55488859).

talbs commented:
@OmarIthawi, thanks for the pull request! I have two questions on the Sass mixin you've added (https://github.com/edx/edx-ora2/pull/664/files#r23714764). Let me know any background information on what this does and why its needed.

OmarIthawi commented:
@talbs and @sarina Thanks for the review :)

Based on your comments, I've updated this PR:
 - Put `direction()` in the correct place `_oa-base.scss`
 - Added comments for the js `gettext` long lines
 - Rebased on top of master to resolve conflicts

sarina commented:
:+1: from me

@talbs can you take another look, and merge if it looks good to you?

talbs commented:
:+1: here as well. 

:truck:ing this fella to Mergetown. Thanks, @OmarIthawi, for the good work.
